### PR TITLE
[#42874] UI for storage authorization state (Backend part)

### PR DESCRIPTION
**Pull request by apfohl**
(Authorize, Error, Connected)

https://community.openproject.org/work_packages/42874

Backend Part.

### CODE REVIEW

File: docs/api/apiv3/components/schemas/storage_model.yml [hunk 1]
```diff
@@ -51,19 +51,33 @@ properties:
...
+              **Resource**: N/A
+      authorize:
+        allOf:
+          - $ref: './link.yml'
+          - description: |-
+              The link to the starting point of the authorization cycle for a configured storage provider.
+              
+              # Conditions
+              
+              `authorizationState` is on of:
```
Comment (wielinde) [inline docs/api/apiv3/components/schemas/storage_model.yml]:
```suggestion
              `authorizationState` is one of:
```

File: docs/api/apiv3/components/schemas/storage_model.yml [hunk 2]
```diff
@@ -51,19 +51,33 @@ properties:
...
+              
+              - urn:openproject-org:api:v3:storages:authorization:FailedAuthorization
+              - urn:openproject-org:api:v3:storages:authorization:Error
```
Comment (wielinde) [inline docs/api/apiv3/components/schemas/storage_model.yml]: @apfohl I disagree: If the authorizationState is Error then there is nothing the user can do, apart from reloading the page. Telling the user to reauthorize when the connection is bad is misleading.

Comment (apfohl) [reply docs/api/apiv3/components/schemas/storage_model.yml]: This is true we can leave it out in this case as well.

**Review by wielinde — APPROVED** [review]
LGTM

**Pull request merged**